src/meta/meta_turtle.c

```diff
@@ -80,129 +92,180 @@ __metadata_backup_target_uri_search(
 }
 
 /*
- * __metadata_load_hot_backup --
- *     Load the contents of any hot backup file.
+ * __wt_read_metadata_file --
+ *     Open metadata file and iterate over the key value pairs calling the worker function for each
+ *     of them.
  */
-static int
-__metadata_load_hot_backup(WT_SESSION_IMPL *session, WT_BACKUPHASH *backuphash)
+int
+__wt_read_metadata_file(WT_SESSION_IMPL *session, const char *file,
+  int (*md_entry_worker_func)(WT_SESSION_IMPL *, WT_ITEM *, WT_ITEM *, void *), void *state,
+  bool *file_exist)
 {
-    WT_CONFIG_ITEM cval;
-    WT_CONNECTION_IMPL *conn;
     WT_DECL_ITEM(key);
     WT_DECL_ITEM(value);
     WT_DECL_RET;
     WT_FSTREAM *fs;
-    size_t allocated_name, file_len, max_len, slot;
-    char *filename, *metadata_conf, *metadata_key, **p, **partial_backup_names, *tablename;
-    const char *drop_cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_drop), "remove_files=false", NULL};
-    bool exist;
 
-    allocated_name = file_len = max_len = slot = 0;
-    conn = S2C(session);
-    filename = metadata_conf = NULL;
-    partial_backup_names = NULL;
-
-    /* Look for a hot backup file: if we find it, load it. */
-    WT_RET(__wt_fs_exist(session, WT_METADATA_BACKUP, &exist));
-    if (!exist)
+    /* Look for a metadata export file: if we find it, load it. */
```

Comment:
This comment is no longer true. I suggest something like:
Look for the given filename. If it exists, load it.

Reply:
Done. Thanks.